### PR TITLE
Generic index runnable

### LINKED ISSUE #88: Generic multispectral index runnable
Add a generic multispectral index runnable that computes an arbitrary multispectral index according to its formula:
- [x] Create a base multispectral index runnable class
- [x] Provide a way to initialize required spectra
- [x] Annotate the index with a generic description
- [x] Turn NDVI into a multispectral index runnable

### PR DESCRIPTION
**Pull request by paulmis**
Implemented the generic index runnable. It reduces code duplication when implementing further indices (e.g. #80). Unfortunately, I don't think the `to_persist` is working (for cloud mode runs), so I commented it out for now. Can you check @mirceatlx?

TODO:
- fix bulk `Mat::get()` and add tests

Closes #88

### CODE REVIEW

File: src/backend/pipeline/modules/index/runnables/generic.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    def prepare(self, data: Data):
+        """
+        Prepares the NDVI data space.
+
+        Args:
+            data: the pipeline data object
+        """
+        super().prepare(data)
+        data.modules[Modules.INDEX.value]["runnables"][self.type.value] = {}
+        # TODO: fix
```
Comment (mirceatlx) [inline src/backend/pipeline/modules/index/runnables/generic.py:80]: We have to fix this issue before merging.

**Comment by mirceatlx** [comment]
You did not change the nutrient index with the new additions. @paulmis

File: src/backend/pipeline/modules/index/runnables/generic.py (hunk 1 above)
Comment (mirceatlx) [reply src/backend/pipeline/modules/index/runnables/generic.py:80]: I managed to solve the problem. However I will let you remove the **TODO** tag after you check it thoroughly. @paulmis

**Comment by paulmis** [comment]
The nutrient index isn't abstractable by `GenericIndex` because it works with masks and patches that the `GenericIndex` doesn't support. The `GenericIndex::calculate()` only takes in the input image rather than any segmentation data.

**Comment by mirceatlx** [comment]
I changed the structure of the Nutrient index such that it can inherit Generic Index. @paulmis take a look

**Review by mirceatlx — APPROVED** [review]

**Pull request merged**